extensions/amp-story/0.1/amp-story-auto-ads.js

```diff
@@ -134,12 +141,43 @@ export class AmpStoryAutoAds extends AMP.BaseElement {
   layoutCallback() {
     return this.ampStory_.signals().whenSignal(CommonSignals.INI_LOAD)
         .then(() => {
+          this.createAdOverlay_();
+          this.initializeListeners_();
           this.readConfig_();
           this.schedulePage_();
         });
   }
 
 
+  /** @private */
+  initializeListeners_() {
+    this.ampStory_.element.addEventListener(EventType.SHOW_AD_UI,
+        this.showAdOverlay_.bind(this));
+    this.ampStory_.element.addEventListener(EventType.HIDE_AD_UI,
+        this.hideAdOverlay_.bind(this));
+  }
+
+
+  /**
+   * show the UI that marks the page as an ad
+   * @private
+   */
+  showAdOverlay_() {
+    if (!this.progressRoot_) {
+      this.progressRoot_ = this.ampStory_.getProgressElement();
+    }
+    this.progressRoot_.style.setProperty('visibility', 'hidden', 'important');
+    this.adLabelElement_.style.visibility = 'visible';
+  }
+
+
+  /** @private */
+  hideAdOverlay_() {
+    this.adLabelElement_.style.visibility = 'hidden';
+    this.progressRoot_.style.setProperty('visibility', 'visible', 'important');
```

Comment:
can we move CSS to .css? And use CSS class or element attribute to control.

Reply:
Yes!